### PR TITLE
Add resize handler to global events and to macro to add event handler functions GetOnresize, SetOnresize

### LINKED ISSUE #7996: Add onresize handler to GlobalEventHandlers
We dispatch the resize event when when the window is resized, but only pages that use `addEventListener` can receive it right now. We should hook up the `document.onresize` mechanism too: https://html.spec.whatwg.org/multipage/webappapis.html#idl-definitions:handler-onresize . This will probably change the output of `./mach test-wpt tests/wpt/web-platform-tests/html/dom/interfaces.html` pass!

Code: components/script/dom/webidls/EventHandler.webidl, components/script/dom/document.rs, components/script/dom/window.rs, components/script/dom/htmlelement.rs

### PR DESCRIPTION
**Pull request by Jeffrharr**
Fixes #7996.

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/8006)

### CODE REVIEW

**Comment by His-name-is-Joof** [comment]
Actually hold on... is this right? Not 100% sure it's firing at the right object.

**Comment by jdm** [comment]
Oh, handy, I forgot that we had this convenient macro :)

**Comment by His-name-is-Joof** [comment]
I think this is what is intended. 

From CSSOM View
When asked to run the resize steps for a Document doc, run these steps:

If doc’s viewport has had its width or height changed (e.g. as a result of the user resizing the browser window, or changing the page zoom scale factor, or an iframe element’s dimensions are changed) since the last time these steps were run, fire an event named resize at the Window object associated with doc.

Assuming that resize is only actually called in the case where something is actually resized and there is no need to check. sending the resize message to the window if it's a document object and normal behavior otherwise.

**Comment by Ms2ger** [comment]
Thank you for your PR; it needs a few code changes (see below).

-S-awaiting-review +S-needs-tests +S-needs-code-changes 


---

Reviewed 1 of 2 files at r1, 4 of 4 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

<sup>**[components/script/dom/document.rs, line 1729 \[r2\]](https://reviewable.io:443/reviews/servo/servo/8006#-K0fz83gPs3m8CxkrBuZ)** ([raw file](https://github.com/servo/servo/blob/b837a14b5cbe3cd4fcc2ba1e83a82723766242f1/components/script/dom/document.rs#L1729)):</sup>
The spec says:

> The following are the event handlers (and their corresponding event handler event types)
> - that must be supported by all HTML elements other than body and frameset elements, as both event handler content attributes and event handler IDL attributes;
> - that must be supported by all Document objects, as event handler IDL attributes; and
> - that must be supported by all Window objects, as event handler IDL attributes on the Window objects themselves, and with corresponding event handler content attributes and event handler IDL attributes exposed on all body and frameset elements that are owned by that Window object's Documents:
> ...

so for `Document`s, the event handler should be registered on the `Document` itself. If you feel up for it, I'd like you to write a test for this; ping me if you need help.


---

<sup>**[components/script/dom/macros.rs, line 277 \[r2\]](https://reviewable.io:443/reviews/servo/servo/8006#-K0fyrjEYXwtq-PDFM-D)** ([raw file](https://github.com/servo/servo/blob/b837a14b5cbe3cd4fcc2ba1e83a82723766242f1/components/script/dom/macros.rs#L277)):</sup>
Drop these unrelated changes, please.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8006)

**Comment by His-name-is-Joof** [comment]
@Ms2ger Week is filled with exams until after Wednesday for me (I'm a senior college student). Hope that's okay. I re-read the DOM and think I understand where I was mixed up and have a local copy ready minus tests. 

I'll ping you when I have some freetime (likely Thursday :( ) to make sure I'm doing things correctly. Where would you expect such a test to go? Part of wpt-tests?

**Comment by jdm** [comment]
Tests in web-platform-tests are organized according to their position in the spec, so https://github.com/w3c/web-platform-tests/tree/master/html/webappapis/scripting/events would be a fine place according to the text at https://html.spec.whatwg.org/multipage/webappapis.html#event-handlers-on-elements,-document-objects,-and-window-objects .

**Comment by Ms2ger** [comment]
Now worries, this is not particularly urgent :)

**Comment by His-name-is-Joof** [comment]
Couldn't find a currently implemented way to force a resize event for testing. Currently using element.dispatchEvent(new Event('resize')), but that needs to be implemented. Probably needs changes, but I'm available now. Thanks a bunch.

**Comment by eefriedman** [comment]
Silly example of forcing a resize event:

```html
<iframe src="data:text/html,<body onresize='document.body.appendChild(document.createTextNode(1));'>"></iframe>
<script>setTimeout(function(){document.querySelector("iframe").width=500;}, 3000);</script>
```

**Comment by Ms2ger** [comment]
The `dispatchEvent` approach is the best here. Could you please squash the commits, though? Feel free to ask on IRC if you need help with that.

**Comment by His-name-is-Joof** [comment]
Oops. ignore that...

**Comment by His-name-is-Joof** [comment]
Okay got it. I'd like to add WPT tests for all of the event handlers (effectively one big loop through them all) although they won't work (or fail) until dispatch event is implemented. I'm fairly unfamiliar with JS, but I figure if I did one, might as well do them all.

If not, I'm sure there are plenty off other issues I could work on.

**Comment by Ms2ger** [comment]
`dispatchEvent` works fine.

**Comment by His-name-is-Joof** [comment]
You are right, you know what they say about assumptions. Your code is wrong and you look like a fool! :P
 
Didn't create WPT test with create-wpt so I didn't add anything to the manifests file. Works correctly now.

**Comment by Ms2ger** [comment]
Sorry for the delay here. I'll try to get back to you asap when the below is fixed.

-S-awaiting-review +S-needs-code-changes 

---

Reviewed 2 of 8 files at r3, 1 of 6 files at r4, 4 of 5 files at r5.
Review status: 5 of 6 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

_[components/script/dom/htmlelement.rs, line 173 \[r3\]](https://reviewable.io:443/reviews/servo/servo/8006#-K3VeMo5ng-13pAmqf4o:-K3VeMo5ng-13pAmqf4p:-180667909) ([raw file](https://github.com/servo/servo/blob/549c5c696f20ccb0a719680cbf307ca8e9e276bb/components/script/dom/htmlelement.rs#L173)):_
This doesn't look right! Can you make sure this is tested?

---


_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8006#-:-K3Vev81myDT68GrZKW7:-1374603763)_

**Comment by frewsxcv** [comment]
@His-name-is-Joof need any help with this?

**Comment by His-name-is-Joof** [comment]
No; it's just been busy and I keep saying, "I really need to do this" then
working on my university courses. Compilers ramped up fast. Finals are in a
week though.

**Comment by His-name-is-Joof** [comment]
I'll go ahead and do this now; it should just need to be made mergable.
Sorry for the delay.

File: tests/wpt/metadata/MANIFEST.json [hunk 1]
```diff
@@ -30141,6 +30141,12 @@
             "url": "/html/semantics/forms/the-input-element/maxlength-manual.html"
           }
         ]
```
Comment (frewsxcv) [inline tests/wpt/metadata/MANIFEST.json]: There might need to be a comma here

**Comment by KiChjang** [comment]
@His-name-is-Joof any updates on this?

**Comment by frewsxcv** [comment]
Closing due to inactivity to let another contributor try their hand at this

**Pull request closed without merging**